#!/usr/bin/env python
PACKAGE = "ta11_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# the three values at the end: default, min, max
gen.add("target_force", double_t, 0, "Threshold force at which controller stops", 0.6, 0.0, 5.0)
gen.add("noise_t", double_t, 0, "noise threshold", 0.04, 0.03, 0.20)
gen.add("k_i", double_t, 0, "PID integral part", 2.8, 0.01, 10.0)
gen.add("k_p", double_t, 0, "PID proportional part", 1.8, 0.01, 3.0)
gen.add("init_k", int_t, 0, "Slope", 1500, 50, 5000)
gen.add("goal_maintain",   bool_t,   0, "Maintain goal force or finish if reached?",  False)

exit(gen.generate(PACKAGE, "ta11_controller", "TA11ControllerDR"))
